novawallet/Common/Web3Names/JsonCanonicalizer.swift

```diff
@@ -0,0 +1,101 @@
+import Foundation
+import OrderedCollections
+
+protocol JsonCanonicalizerProtocol {
+    func canonicalizeJSON(_ data: Data) throws -> String?
+}
+
+final class JsonCanonicalizer: JsonCanonicalizerProtocol {
+    func canonicalizeJSON(_ data: Data) throws -> String? {
+        let jsonObject = try JSONSerialization.jsonObject(with: data, options: [])
+        let canonicalizedObject = canonicalize(jsonObject)
+        return canonicalizedObject
+    }
+
+    private func canonicalize(_ object: Any) -> String {
+        var result = ""
+
+        if let string = object as? String {
+            result += "\"\(canonicalize(string: string))\""
+        } else if let array = object as? [Any] {
+            result += "["
+            var next = false
+            array.forEach { element in
+                if next {
+                    result += ","
+                }
+                next = true
+                result += canonicalize(element)
+            }
+            result += "]"
+        } else if let dictionary = object as? [String: Any] {
+            result += "{"
+            var next = false
+            dictionary.keys.sorted { (str1, str2) -> Bool in
+                let utfView1 = Array(str1.utf16)
+                let utfView2 = Array(str2.utf16)
+
+                for index in 0 ..< max(utfView1.count, utfView2.count) {
+                    if let char1 = utfView1[safe: index], let char2 = utfView2[safe: index] {
+                        if char1 == char2 {
```

Comment:
I think more cleaner way is to just check for inequality. In this case we don't need to use ```continue```.